### PR TITLE
[IMP] point_of_sale: remove the background loading of partners and products

### PR DESCRIPTION
**Pull request by adgu-odoo**
The loading of products and partners in background is too slow for the moment. For configurations with a lot of products or partners, the loading is slowing the whole behaviour of the PoS while it is not really necessary to load 20K products. This PR removes the loading of partners and products in background and add a button on the ProductScreen for user to load more products when a search term is put in the search bar. This load will load 30 items at a time.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/models/res_config_settings.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def _default_pos_config(self):
     pos_picking_policy = fields.Selection(related='pos_config_id.picking_policy', readonly=False)
     pos_picking_type_id = fields.Many2one(related='pos_config_id.picking_type_id', readonly=False)
     pos_pricelist_id = fields.Many2one('product.pricelist', string='Default Pricelist', compute='_compute_pos_pricelist_id', readonly=False, store=True)
-    pos_product_load_background = fields.Boolean(related='pos_config_id.product_load_background', readonly=False)
+    pos_partner_load_background = fields.Boolean(related='pos_config_id.partner_load_background', readonly=False)
```
Comment (KangOl) [inline addons/point_of_sale/models/res_config_settings.py:89]: revert this change

File: addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js [hunk 2]
```diff
@@ -173,13 +171,25 @@ odoo.define('point_of_sale.PartnerListScreen', function(require) {
...
             let domain = [];
+            const limit = 30;
+            let offset = this.state.currentOffset;
             if(this.state.query) {
                 domain = ['|', ["name", "ilike", this.state.query + "%"],
                                ["parent_name", "ilike", this.state.query + "%"]];
+
+                if (this.state.previousQuery != this.state.query) {
+                    offset = 0;
+                }
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js:192]: history since we talked on discord: maybe reset the `currentOffset` in the searchPartner

File: addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidget.js [hunk 3]
```diff
@@ -75,6 +77,66 @@ odoo.define('point_of_sale.ProductsWidget', function(require) {
...
+        async loadProductFromDB() {
+            if(!this.state.searchWord)
+                return;
+
+            try {
+                const limit = 30;
+                let offset = this.state.currentOffset;
+                if (this.state.previousSearchWord != this.state.searchWord) {
+                    offset = 0;
+                }
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidget.js:109]: same here maybe reset offset in the search

**Comment by trj-odoo** [comment]
nice work
@robodoo r+
